### PR TITLE
[New] Add event-drops w/ npm auto-update via single package.json

### PR DESCRIPTION
**Pull request by VarunWachaspati**
Pull request for issue: #11999
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/marmelab/EventDrops
 * Official website (optional, not the repository): http://marmelab.com/EventDrops/
 * NPM package url (optional): https://www.npmjs.com/package/event-drops
 * License and its reference: MIT License
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 40
   - Count of stars: 792
   - Count of forks: 157
 * NPM download stats (optional):
   - Downloads in the last day: 3
   - Downloads in the last week: 74
   - Downloads in the last month: 284

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/event-drops/package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+	"name": "event-drops",
+	"filename": "src/index.min.js",
+	"version": "0.3.1-alpha3",
```
Comment (PeterDaveHello) [inline ajax/libs/event-drops/package.json]: Please remove its version info here.

File: ajax/libs/event-drops/package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	"description": "A time based / event series interactive visualization using d3.js. Use drag and zoom to navigate in time.",
+	"main": "dist/index.js",
```
Comment (PeterDaveHello) [inline ajax/libs/event-drops/package.json]: We don't need "main" here

File: ajax/libs/event-drops/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          "meatballs.js",
+          "xAxis.js",
+          "zoom.js",
+          "style.css",
+          "drawer/delimiters.js",
+          "drawer/drops.js",
+          "drawer/index.js",
+          "drawer/labels.js",
+          "drawer/lineSeparator.js",
+          "drawer/xAxis.js"
```
Comment (PeterDaveHello) [inline ajax/libs/event-drops/package.json]: Do we need all these files? I think we should use npm auto-update here and grab the whole `dist` folder.

File: ajax/libs/event-drops/package.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{
+	"name": "event-drops",
+	"filename": "src/index.min.js",
```
Comment (PeterDaveHello) [inline ajax/libs/event-drops/package.json]: Since `src` is the basePath, it won't show up in the filename.

**Comment by VarunWachaspati** [comment]
Hey @PeterDaveHello 
I am not able to understand the CI error now, could you please help me out?

**Comment by PeterDaveHello** [comment]
@VarunWachaspati sure thing

**Comment by PeterDaveHello** [comment]
@VarunWachaspati you need to add some keywords for it, I believe that you can learn from jquery's package.json: https://github.com/cdnjs/cdnjs/blob/master/ajax/libs/jquery/package.json, it'll be better to use good-looking indent, and you'll also need to use npm auto-update as jquery did :)

**Comment by VarunWachaspati** [comment]
@PeterDaveHello passed all tests, Thanks for the help!

File: ajax/libs/event-drops/package.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+	"npmName": "event-drops",
+	"npmFileMap": [
+		{
+		  "basePath": "src",
+		  "files": [
+	      "**/*"
+	    ]
+		}
+	],
+	"filename": "index.min.js",
```
Comment (austin665) [inline ajax/libs/event-drops/package.json]: Should this be eventDrops.min.js instead ?

File: ajax/libs/event-drops/package.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{
+	"name": "event-drops",
+	"npmName": "event-drops",
+	"npmFileMap": [
+		{
+		  "basePath": "src",
```
Comment (maruilian11) [inline ajax/libs/event-drops/package.json]: Please set `dist` as basePath instead. Files in `dist` are what we need.

File: ajax/libs/event-drops/package.json (hunk 5 above)
Comment (maruilian11) [inline ajax/libs/event-drops/package.json]: Please use `eventDrops.js` instead.

Comment (VarunWachaspati) [reply ajax/libs/event-drops/package.json]: There is no file named "eventDrops.js" in the repository hence I added index.js instead
Let me know if it's still wrong I will gladly make the changes

File: ajax/libs/event-drops/package.json (hunk 6 above)
Comment (VarunWachaspati) [reply ajax/libs/event-drops/package.json]: Hi, 
In the Github repository there is no "dist" directory hence I included "src" directory as basePath
If it is still wrong then I would gladly change it to "dist"

**Comment by VarunWachaspati** [comment]
@PeterDaveHello Could you please help me rebase this as well? 
Thanks in Advance

File: ajax/libs/event-drops/package.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "event-drops",
+  "filename": "eventDrops.min.js",
```
Comment (PeterDaveHello) [inline ajax/libs/event-drops/package.json]: This should be `eventDrops.js`, not `eventDrops.min.js`, as it's already minified.

**Comment by sufuf3** [comment]
@VarunWachaspati thanks for contribution.
Would you like to work on this and help improve this PR? 
Just modified the `"filename"` field from `"eventDrops.min.js"` to `"eventDrops.js"`

Many thanks.

**Comment by PeterDaveHello** [comment]
Close this PR due to no response. Feel free to open a new one, please.

**Comment by VarunWachaspati** [comment]
@PeterDaveHello @sufuf3 Sorry for the delay, made the change you requested 
Could you please rebase the branch for me as it is quite old?
Thanks in Advance

**Comment by PeterDaveHello** [comment]
@VarunWachaspati Sure.

**Pull request merged**